hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/file/TestOMFileCreateRequestWithFSO.java

```diff
@@ -156,6 +157,18 @@ public void testValidateAndUpdateCacheWithNonRecursiveAndOverWrite()
     testNonRecursivePath(key, false, false, true);
   }
 
+  @Test
+  public void testValidateAndUpdateCacheWithSnapshotReservedWord()
+      throws Exception {
+    String[] validPaths = {
+        "a/b/" + OM_SNAPSHOT_INDICATOR + "c/d",
+        "a/b/c/" + OM_SNAPSHOT_INDICATOR + "/d"
```

Comment:
For the first two levels of path ("a/b/c/d"), ozone assumes the first level (a) as volume, the second level (b) as bucket, rest of levels are directory and the last level is file name.